internal/services/web/function_app.go

```diff
@@ -356,7 +350,7 @@ func getBasicFunctionAppAppSettings(d *pluginsdk.ResourceData, appServiceTier, e
 	// On consumption and premium plans include WEBSITE_CONTENT components, unless it's a Linux consumption plan
 	// (see https://github.com/Azure/azure-functions-python-worker/issues/598)
 	if !(strings.EqualFold(appServiceTier, "dynamic") && strings.EqualFold(d.Get("os_type").(string), "linux")) &&
-		(strings.EqualFold(appServiceTier, "dynamic") || strings.HasPrefix(strings.ToLower(appServiceTier), "elastic")) {
```

Comment:
Just looking to confirm that we don't need to check for `elastic` anymore? Is there any chance someone has that still?

Reply:
As far as I know, Azure Files is only required for Consumption (Windows) and Elastic Premium (Windows / Linux). I think it is correct to leave this conditional expression as `elastic`. See also https://github.com/hashicorp/terraform-provider-azurerm/pull/13218#issuecomment-913755737

Reply:
Hi both, from the [docs](https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings#website_contentshare):
> Only used when deploying to a Premium plan or to a Consumption plan running on Windows. Not supported for Consumptions plans running Linux.

I initially misinterpreted this as ElasticPremium also, however, this is actually App Service Premium Plans. For context see #13566.

Reply:
This is very confusing, but the Premium V2 / V3 of the App Service Plan and the Azure Functions Premium Plan are completely different services. The Premium Plan in this document refers to the latter.

If the File Share setting is added to the App Service Plan (Premium V2 / V3), the current working application may be lost. This is a very dangerous change and the application I manage is also affected.

Reply:
> This is very confusing,

You are not wrong there. 🙃 

The resource has always previously sent the content details for premium SKU's, and elasticpremium plans [see here](https://github.com/hashicorp/terraform-provider-azurerm/blob/v2.76.0/internal/services/web/function_app.go#L334)

The change in 2.77 was intended to align the resource behaviour with the docs, but appears to have created the issue in 13218 where these settings were being sent where they had previously not been. Are you reporting that the Premium plans did _not_ get these settings as intended?


The other change in this PR addresses the change of setting for the share directory, which is a bug in the 2.77 implementation which attempted to maintain any existing value configured. (see 13536) If the property already exists in the Function App AppSettings, then it will be used and not added/created. Additionally this is intended to address a related bug whereby the provider created the share with a `-content` suffix that prevented proper operation of slots. This will also mean that if the share property is set for a Premium Plan, it will be honoured/maintained and not overwritten.

Reply:
As for #13566, it is different from this issue.

Azure Functions has an option to operate without File Share, so I think it's more appropriate to add a property (e.g. `disabled_builtin_file_share` or etc) to not auto-set `WEBSITE_CONTENTSHARE` and `WEBSITE_CONTENTAZUREFILECONNECTIONSTRING`.

https://docs.microsoft.com/en-us/azure/azure-functions/storage-considerations#create-an-app-without-azure-files

Reply:
Thanks for your insight @shibayan - I'm taking this back to the drawing board to work it back through...

Reply:
I've updated to correct the behaviour for PremiumV2/V3 (i.e. not send) and added the reuse of an existing value if defined (in case users do have it defined).  I believe this is going to be sufficient for this stage, as presently this resource is not usable since 2.77 because of the share recreation bug. I'm reticent to introduce another property at this stage that will further affect behaviour of the resource. Upon re-reading, I believe 13566 to be a symptom of the share recreation bug, so should be resolved by this PR in any case.

fwiw - I'm in the process of completely rewriting this resource for the beta / v3.0, so I've made notes for addressing this properly there. I'll also be speaking to the Service Team on this as soon as possible.

Reply:
Thank you for the quick response. I've checked the changed code and I'm sure it works ideally with Provider v2.x.